com.unity.render-pipelines.high-definition/Editor/Material/Decal/DecalProjectorEditor.cs

```diff
@@ -363,22 +398,31 @@ void DrawUVHandles(DecalProjector decalProjector)
                 {
                     Undo.RecordObject(decalProjector, "Decal Projector Change");
 
-                    Vector2 limit = new Vector2(Mathf.Abs(decalProjector.size.x * k_LimitInv), Mathf.Abs(decalProjector.size.y * k_LimitInv));
-                    Vector2 uvScale = uvHandles.size;
                     for (int channel = 0; channel < 2; channel++)
                     {
-                        if (Mathf.Abs(uvScale[channel]) > limit[channel])
-                            uvScale[channel] = decalProjector.size[channel] / uvScale[channel];
-                        else
-                            uvScale[channel] = Mathf.Sign(decalProjector.size[channel]) * Mathf.Sign(uvScale[channel]) * k_Limit;
+                        // Preserve serialized state for axes with the scaled size 0.
+                        if (scaledSize[channel] != 0f)
+                        {
+                            float handleSize = uvHandles.size[channel];
+                            float minusNewUVStart = .5f * handleSize - uvHandles.center[channel];
+                            float decalSize = decalProjector.size[channel];
+                            float limit = k_LimitInv * decalSize;
+                            if (handleSize > limit || handleSize < -limit)
+                            {
+                                uvScale[channel] = decalSize / handleSize;
+                                uvBias[channel] = minusNewUVStart / handleSize;
+                            }
+                            else
+                            {
+                                // TODO: Decide if decalProjector.size and uvHandles.size should ever have negative values. They can't currently.
```

Comment:
I don't think it is a good Iidea. As you said there is already a missing negative UV. I see no interest to have both.

Reply:
Can you clarify: Not a good idea to ever have negative values in these properties? Or not a good idea to remove the multiplication by their signs, just in case they will become negative someday?

I can imagine the need in negative UV tiling that may require allowing negative size in these handles.

Reply:
no I mean: if we add the negative value for UV and we will try as it can be usefull, I don't see interest to also have negative value for size. Seams kind of a useless complication to me.

Why negative UV tilling should recquire negative size in handles? I think I miss something here.

Reply:
When UV tiling is positive and user would want to flip it with handles by moving an edge from one side to another the movement will stop at the zero rect size, won't it? Because uvHandles.size will ensure that its size after editing is non-negative.
https://github.com/Unity-Technologies/Graphics/blob/848d723429c6c35fe19c12350a44a8b55b2b757d/com.unity.render-pipelines.high-definition/Editor/Material/Decal/DisplacableRectHandles.cs#L288

Reply:
Oh right sorry I was considering the wrong handle from the beginning. Yes totally make sens.

Reply:
So I can remove `* Mathf.Sign(decalSize)` but keep `* Mathf.Sign(handleSize)`, and remove a todo, right?

Reply:
Well If I am not wrong, the DisplaceableRectHandles will not be able to display negative size. It needs a little of work on this too for the handle to works correctly. So for now, I assume that keeping the comment is better and we will check this along the negative UV request. What do you think?

(If I am wrong and still work please do it)

Reply:
Yes, keeping a comment until the negative UV request makes sense. I'll only remove the decal size sign multiplication for now.